src/openms/source/ANALYSIS/TOPDOWN/FLASHDeconvHelperStructs.cpp

```diff
@@ -0,0 +1,345 @@
+  }
+
+  Size FLASHDeconvHelperStructs::PrecalculatedAveragine::getApexIndex(const double mass) const
+  {
+    return apex_index_[massToIndex_(mass)];
+  }
+
+  Size FLASHDeconvHelperStructs::PrecalculatedAveragine::getLastIndex(const double mass) const
+  {
+    Size index = massToIndex_(mass);
+    return apex_index_[index] + right_count_from_apex_[index];
+  }
+
+
+  void FLASHDeconvHelperStructs::PrecalculatedAveragine::setMaxIsotopeIndex(const int index)
+  {
+    max_isotope_index_ = index;
+  }
+
+  FLASHDeconvHelperStructs::LogMzPeak::LogMzPeak(const Peak1D& peak, const bool positive) :
+      mz(peak.getMZ()),
+      intensity(peak.getIntensity()),
+      logMz(getLogMz(peak.getMZ(), positive)),
+      abs_charge(0),
+      is_positive(positive),
+      isotopeIndex(0)
+  {
+  }
+
+  double FLASHDeconvHelperStructs::LogMzPeak::getUnchargedMass()
+  {
+    if (abs_charge == 0)
+    {
+      return .0;
+    }
+    if (mass <= 0)
+    {
+      mass = (mz - getChargeMass(is_positive)) * abs_charge;
+    }
+    return mass;
+  }
+
+  bool FLASHDeconvHelperStructs::LogMzPeak::operator<(const LogMzPeak& a) const
+  {
+    if(this->logMz == a.logMz)
+    {
+      return this->intensity < a.intensity;
+    }
+    return this->logMz < a.logMz;
+  }
+
+  bool FLASHDeconvHelperStructs::LogMzPeak::operator>(const LogMzPeak& a) const
+  {
+    if(this->logMz == a.logMz)
+    {
+      return this->intensity > a.intensity;
+    }
+    return this->logMz > a.logMz;
+  }
+
+  bool FLASHDeconvHelperStructs::LogMzPeak::operator==(const LogMzPeak& a) const
+  {
+    return this->logMz == a.logMz && this->intensity == a.intensity;
+  }
+
+  double FLASHDeconvHelperStructs::getChargeMass(const bool positive)
+  {
+    return (positive ? Constants::PROTON_MASS_U : -Constants::PROTON_MASS_U);
+  }
+
+
+  double FLASHDeconvHelperStructs::getLogMz(const double mz, const bool positive)
+  {
+    return std::log(mz - getChargeMass(positive));
+  }
+
+
+  FLASHDeconvHelperStructs::TopPicItem::TopPicItem(String in)
+  {
+    str = in;
+    std::vector<String> results;
+    std::stringstream tmp_stream(in);
+    String str;
+
+    while (getline(tmp_stream, str, '\t'))
+    {
+      results.push_back(str);
+    }
+    prsm_id = stoi(results[1]);
+    spec_id = stoi(results[2]);
+    scan = stoi(results[4]);
+    rt = stod(results[5]);
+    peak_count = stoi(results[6]);
+    charge = stoi(results[7]);
+    precursor_mass = stod(results[8]);
+    adj_precursor_mass = stod(results[9]);
+    proteform_id = stoi(results[10]);
+    if (results[11].hasPrefix("-"))
+    {
+      intensity = 0;
+    }
+    else
+    {
+      intensity = stod(results[11]);
+    }
+    String acc = results[13];
+    int first = acc.find("|");
+    int second = acc.find("|", first + 1);
+    protein_acc = acc.substr(first + 1, second - first - 1);
+    first_residue = stoi(results[15]);
+    last_residue = stoi(results[16]);
+    if (stoi(results[18]) == 0)
+    {
+      //unexp_mod = .0;
```

Comment:
This should either be changed to `continue;` or otherwise clarified.

Reply:
Changed to

if (stoi(results[18]) != 0)
{

and removed 

if 
..
else
{